astropy/time/core.py

```diff
@@ -593,6 +607,8 @@ def _get_time_fmt(self, val, val2, format, scale, precision, in_subfmt, out_subf
         problems = {}
         for name, cls in formats:
             try:
+                if np.any(mask):
+                    val, val2 = cls._fill_masked_values(val, val2, mask, in_subfmt)
```

Comment:
This seems a bit risky, as now a badly written `_fill_masked_values` for one format can cause problems for the next one; I'll revise.